GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs

```diff
@@ -102,5 +122,27 @@ public IEnumerable<Case> GetCases(ulong projectId, ulong suiteId)
             ValidateRequestResult(result);
             return result.Payload;
         }
+
+        /// <summary>
+        /// Pauses sending of requests when requests per minutes limit is reached 
+        /// </summary>
+        /// <param name="elapsedMilliSeconds">Milliseconds elapsed from start</param>
+        public void RequestsLimitCheck(double elapsedMilliSeconds)
+        {
+            //To ensure that limiter works even when one or more minute has passed
+            elapsedMilliSeconds %= 60_000;
+            if (RequestsCount + 1 >= _config.TestRailMaxRequestsPerMinute &&
+                elapsedMilliSeconds <= 59_000)
+            {
+                //additional 3000 milliseconds of sleep - just in case
+                var sleepTime = (int) Math.Round(63_000 - elapsedMilliSeconds);
+                Log.Debug($"Limit of {_config.TestRailMaxRequestsPerMinute} requests per minute is reached. Waiting for {sleepTime} seconds to continue...");
+                Thread.Sleep(sleepTime);
+                RequestsCount = 0;
+                Log.Debug($"Waiting completed. Requests count set to 0");
+            }
+            else if (RequestsCount>=180) 
```

Comment:
Why do we need this condition?

Reply:
We need this to set RequestCount to 0 when both timer haven't reached 60 seconds and RequestsCount does not pass the threshold, so in the next 60 seconds we will start counting requests from zero again